Branch-SDK/Branch-SDK/Core/BNCStrongMatchHelper.m

```diff
@@ -0,0 +1,104 @@
+//
+//  BNCStrongMatchHelper.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 8/26/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BNCStrongMatchHelper.h"
+#import <SafariServices/SafariServices.h>
+#import "BNCConfig.h"
+#import "BNCPreferenceHelper.h"
+#import "BNCSystemObserver.h"
+#import "BranchConstants.h"
+
+#define BRANCH_STRONG_MATCH_CREATED_DATE @"bnc_strong_match_created_date"
+
+@interface BNCStrongMatchHelper () <SFSafariViewControllerDelegate>
+@property (nonatomic, strong) UIWindow *secondWindow;
+@property (nonatomic) BOOL requestInProgress;
+@end
+
+@implementation BNCStrongMatchHelper
+
++ (BNCStrongMatchHelper *)strongMatchHelper {
+    static BNCStrongMatchHelper *strongMatchHelper;
+    static dispatch_once_t onceToken;
+    
+    dispatch_once(&onceToken, ^{
+        strongMatchHelper = [[BNCStrongMatchHelper alloc] init];
+    });
+    
+    return strongMatchHelper;
+}
+
+- (void)createStrongMatchWithBranchKey:(NSString *)branchKey {
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000
+    if (_requestInProgress) {
+        return;
+    }
+    _requestInProgress = YES;
+    
+    NSDate *thirtyDaysAgo = [NSDate dateWithTimeIntervalSinceNow:-(60*60*24*30)];
+    NSDate *lastCheck = [[NSUserDefaults standardUserDefaults] objectForKey:BRANCH_STRONG_MATCH_CREATED_DATE];
+    if ([lastCheck compare:thirtyDaysAgo] == NSOrderedDescending) {
+        _requestInProgress = NO;
+        return;
+    }
+    
+    [self presentSafariVCWithBranchKey:branchKey];
+#endif
+}
+
+- (void)presentSafariVCWithBranchKey:(NSString *)branchKey {
+    NSString *urlString = [NSString stringWithFormat:@"%@/_strong_match?os=iOS", BNC_LINK_URL];
+    
+    BNCPreferenceHelper *preferenceHelper = [BNCPreferenceHelper preferenceHelper];
+    BOOL isRealHardwareId;
+    NSString *hardwareId = [BNCSystemObserver getUniqueHardwareId:&isRealHardwareId andIsDebug:preferenceHelper.isDebug];
+    if (!hardwareId || !isRealHardwareId) {
+        _requestInProgress = NO;
+        return;
+    }
+    
+    if (preferenceHelper.deviceFingerprintID) {
+        urlString = [urlString stringByAppendingString:[NSString stringWithFormat:@"&%@=%@", BRANCH_REQUEST_KEY_DEVICE_FINGERPRINT_ID, preferenceHelper.deviceFingerprintID]];
+        
+    } else {
+        urlString = [urlString stringByAppendingString:[NSString stringWithFormat:@"&%@=%@", BRANCH_REQUEST_KEY_HARDWARE_ID, hardwareId]];
```

Comment:
Let's add this in 100% of the cases. This is in case we later decide to start checking if the hardware id changed for a device.